src/model_theory/terms_and_formulas.lean

```diff
@@ -286,14 +305,22 @@ def exs : ∀ {n}, L.bounded_formula α n → L.formula α
 | 0 φ := φ
 | (n + 1) φ := φ.ex.exs
 
+/-- Raises all of the `fin`-indexed variables of a formula greater than or equal to `m` by `n'`. -/
+def lift_at : ∀ {n : ℕ} (n' m : ℕ), L.bounded_formula α n → L.bounded_formula α (n + n')
+| n n' m falsum := falsum
+| n n' m (equal t₁ t₂) := (t₁.lift_at n' m).bd_equal (t₂.lift_at n' m)
+| n n' m (rel R ts) := R.bounded_formula (term.lift_at n' m ∘ ts)
+| n n' m (imp f₁ f₂) := (f₁.lift_at n' m).imp (f₂.lift_at n' m)
+| n n' m (all f) := ((f.lift_at n' m).cast (by rw [add_assoc, add_comm 1, ← add_assoc])).all
```

Comment:
At some point, I ended up switching an order convention - I should either make a detailed note of this, or switch it back. Specifically, my `all` quantifier applies to the last, instead of the first, variable in the finite tuple - something you cannot do when you have a countable sequence of them. This does have the advantage of keeping most of the variable names the same when you quantify.

Reply:
Oh, interesting. Yeah, that should work, and then this is what you want. 
I remember now that you're using `fin.snoc` in `bounded_formula.realize`.
Could you document this in the module doc?

Reply:
I've left a note in the module doc.